docs/api/memory/link_time_optimization.md

```diff
@@ -0,0 +1,29 @@
+# Link Time Optimization
+
+Link Time Optimization (LTO) is a program memory usage optimization mechanism performed by compiler at link time. At compile time compiler creates special intermediate represention of all translation units and then optimize them as a single unit at link time resulting in much better optimization comparing to non-LTO builds.
+
+## Using LTO in Mbed OS
+
+In Mbed OS build system LTO is implemented as an optional profile extension placed in `tools\profiles\extensions\lto.json`. When enabled, build profile is amended with LTO flags. LTO perform heavy memory optimizations what breaks debugging, so it's not recommended to enable it in debug/develop profiles.
+
+To enable LTO retype `--profile` option with LTO file path `tools\profiles\extensions\lto.json`
+
+<span class="notes">**Note**: For profile extensions you have to put full path relative to Mbed OS root folder.</span>
+
+To enable LTO together with `release` profile:
+```
+mbed compile -t TOOLCHAIN -m TARGET --profile release --profile tools/profiles/extensions/lto.json
+```
+
+<span class="notes">**Note**: In LTO builds compiler produce bytecode/bitcode instead of regular object code. And it's hard to analyse this output by object code analysis tools.</span>
+
+## Limitations
```

Comment:
Also, I suggest in the Limitation section to mention the impact to the debuggability also the impact to the symbol size, which disruption to memory map generation.  even we briefly mentioned above. here is an article that could be useful https://interrupt.memfault.com/blog/best-and-worst-gcc-clang-compiler-flags#-flto

Reply:
updated